File: solutions/go/leap/1/leap.go
// Package leap provides functionality for detecting leap years.
package leap

// IsLeapYear returns true if the given year is a leap year.
func IsLeapYear(year int) bool {
	return (year%100 == 0 && year%400 == 0) ||
		(year%4 == 0 && year%100 != 0)
}

File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"

namespace leap {
  bool is_leap_year(const int year) {
    // on every year that is evenly divisible by 4
    //   except every year that is evenly divisible by 100
    //     unless the year is also evenly divisible by 400
    if (year % 100 == 0 && year % 400 != 0) return false;
    return (year % 4 == 0);
  }
}